RELEASE.md

```diff
@@ -17,6 +17,64 @@ documentation].
 
 [the release documentation]: https://gateway-api.sigs.k8s.io/concepts/versioning/
 
+## Releasing a monthly version
+
+### Starting point
+
+Make sure all the changes that should be part of the monthly release are
+merged into `main`.
+
+### Tagging the monthly release
+
+Start by tagging the `main` branch with a tag of the form
+`monthly-YYYY.MM` (for example, `monthly-2025.11`). Push this to GitHub.
+Later, CI will handle the rest of the release process -- for now, you'll
+do it by hand.
+
+### Building the monthly YAMLs
+
+```
+make build-monthly-yaml
+```
+
+This will use the current date to generate a tag of the form
+`monthly-YYYY.MM` (for example, `monthly-2025.11`), then generate the
+install YAMLs for the monthly release in `release/$TAG-install.yaml`. If
+desired, you can override the tag by setting `TAG` in your environment
+before running the command.
+
+Generating the monthly CRD YAML makes changes in the
+`config/crd/experimental/` directory. These changes should not be
+committed, so `make build-monthly-yaml` will ask if you want it to use
+`git restore` to clean up. Since these are derived objects, typically you
+would answer "Y" to this question. To prevent the prompt and silently
+clean up, set `MONTHLY_NONINTERACTIVE=true` in your environment before
+running the command.
+
+Once you've generated the monthly YAML, you can create a GitHub release
+and attach the `release/$TAG-install.yaml` file to it as an artifact.
+**DO NOT** mark this release as the latest release.
+
+### Writing the Release Changelog
+
+Since monthlies always happen on `main` and affect only the experimental
+resources, there's a streamlined process for generating the changelog.
+
+You'll need the `$TAG` used for this monthly release as well as `$PREV_TAG` used for the previous monthly release, then:
+
+```
+git log --stat ${PREV_TAG}..${TAG} -- config/crd/experimental
+```
+
+will show you a one-line summary of all the changes to experimental CRDs
```

Comment:
Do we want to limit this to CRD changes only? Our other releases have a more inclusive changelog. Maybe monthly releases should just use GitHub's autogenerated changelog for releases?

Reply:
I mean, you'll get the autogenerated log anyway, the point here is providing some assistance for having a human-readable version too.

Given that the only artifact we're actually going to release is the CRD YAML, including things like e.g. changes to the conformance suite seems strange to me, though I could be convinced otherwise?

Reply:
I think maybe a note saying "it's okay to just have the autogenerated changelog if there's nothing major". Since the release _can_ be used for testing updated conformance tests, when used as a tag for a Go module or similar.